internal/distributed/querynode/client/client.go

```diff
@@ -64,7 +65,9 @@ func (c *Client) connect(retryOptions ...retry.Option) error {
 	connectGrpcFunc := func() error {
 		opts := trace.GetInterceptorOpts()
 		log.Debug("QueryNodeClient try connect ", zap.String("address", c.addr))
-		conn, err := grpc.DialContext(c.ctx, c.addr,
+		ctx, cancel := context.WithTimeout(c.ctx, 15*time.Second)
+		defer cancel()
```

Comment:
Is connect immediately disconnected after creation?

Reply:
No, the cancel() controls the connection process and won't affect subsequent operations